tests/ui/type/type-check/assignment-in-if.stderr

```diff
@@ -67,9 +67,6 @@ LL |             x == 5
 error[E0308]: mismatched types
   --> $DIR/assignment-in-if.rs:44:18
    |
-LL |     if y = (Foo { foo: x }) {
-   |                        - here the type of `x` is inferred to be `usize`
```

Comment:
These suggestions are nonsense because `x` started out as `{integer}`, which was already incompatible with `bool`, which is the expectation of the E0308.